### PR TITLE
[Fixes #80] Formatter shouldn't add exception for overloads where exc…

### LINKED ISSUE #80: Formatter shouldn't add exception for overloads where exception isn't passed in
Specific example,
```
        private static void Write(
            this ILogger logger,
            LogLevel logLevel,
            ILoggerStructure state,
            Exception exception = null)
        {
            logger.Write(logLevel, 0, state, exception, _loggerStructureFormatter);
        }
```
the formatter there always adds a newline and the exception, when it should just be the ILS formatted message

### PR DESCRIPTION
**Pull request by kichalla**
…eption isn't passed in

@Eilon @BrennanConroy

### CODE REVIEW

File: src/Microsoft.Framework.Logging.Abstractions/LoggerExtensions.cs [hunk 1]
```diff
@@ -581,17 +575,17 @@ public static void LogCritical([NotNull] this ILogger logger, int eventId, strin
...
-        private static string LogValuesFormatter(ILogValues state, Exception exception)
+        private static string MessageFormatter(object state, Exception exception)
         {
             if (exception == null)
             {
                 return state?.ToString();
             }
 
-            return state?.ToString() + Environment.NewLine + exception;
+            return string.Format(CultureInfo.CurrentCulture, "{0}{1}{2}", state, Environment.NewLine, exception);
```
Comment (BrennanConroy) [inline src/Microsoft.Framework.Logging.Abstractions/LoggerExtensions.cs]: Is state no longer able to be null now?

Comment (kichalla) [inline src/Microsoft.Framework.Logging.Abstractions/LoggerExtensions.cs]: good question...state being null is like the log message being null...yeah, a log statement without a message is unusual but definitely possible(scenario being: an exception is caught and user just wants to log the exception and does not want to provide any message)...I was initially thinking of why not have NotNull attributes on all the extensions where we accept `state`, but the above scenario invalidates it... @Eilon, thoughts?

Comment (Eilon) [inline src/Microsoft.Framework.Logging.Abstractions/LoggerExtensions.cs]: I would think that `state` could definitely be null, no? I bet the typical quick & dirty logging pattern is to just have a log level and a message.

Comment (kichalla) [inline src/Microsoft.Framework.Logging.Abstractions/LoggerExtensions.cs]: > I bet the typical quick & dirty logging pattern is to just have a log level and a message.

Here `state` represents the message actually...so did you mean to say 'is to just have a log level and an exception'?

Comment (Eilon) [inline src/Microsoft.Framework.Logging.Abstractions/LoggerExtensions.cs]: Oh, state is a message? Well that's nice :smile: I really did mean log level + message. Imagine a call to `Logger.LogWarning("blah blah bad thing happened")` - that's super common. But not message? Maybe less common...

Comment (kichalla) [inline src/Microsoft.Framework.Logging.Abstractions/LoggerExtensions.cs]: I hope I understood what you meant in the last comment, but I have updated the PR now. please take a look /cc: @Eilon @BrennanConroy

**Comment by BrennanConroy** [comment]
Looks good to me
:shipit:

**Comment by kichalla** [comment]
commit 73bd91bbe9aebe74743e3215ac0dbb5d6612320e

**Pull request closed without merging**